### ISSUE TITLE
Ignored folder is not ignored

### ISSUE THREAD

**Issue by bodydisplaynone** [microsoft/vscode#169678]
Type: <b>Bug</b>

Not sure what's happening here but seems like my ignored folders are actually NOT ignored by git. The first obvious one was the .vscode directory that is added automatically as default but doesn't matter what I add here it still gets listed as tracked.

I have another machine with the same settings (sync) and that one works perfectly. Any idea what's going on? I tried removing git cache, tried different regex patterns but nothing worked so far.

![image](https://user-images.githubusercontent.com/22660372/208761084-87498359-bd06-446d-b240-31646e5dbf85.png)

VS Code version: Code 1.74.2 (e8a3071ea4344d9d48ef8a4df2c097372b0c5161, 2022-12-20T10:29:14.590Z)
OS version: Windows_NT x64 10.0.22621
Modes:
Sandboxed: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-10750H CPU @ 2.60GHz (12 x 2592)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|undefined|
|Memory (System)|31.87GB (22.72GB free)|
|Process Argv|--crash-reporter-id 9e80de22-9a8a-4654-872e-69670f9cd8ec|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (50)</summary>

Extension|Author (truncated)|Version
---|---|---
better-comments|aar|3.0.2
html-end-tag-labels|ant|0.9.0
vscode-color-picker|Ant|0.0.4
quitcontrol-vscode|art|4.0.0
vscode-custom-css|be5|7.1.0
vscode-log-viewer|ber|0.12.2
vscode-intelephense-client|bme|1.8.2
simple-react-snippets|bur|1.2.7
run-cy|cor|0.1.1
vscode-mysql-client2|cwe|6.0.4
dart-code|Dar|3.54.1
flutter|Dar|3.54.0
vscode-eslint|dba|2.2.6
gitlens|eam|13.1.1
prettier-vscode|esb|9.10.3
auto-close-tag|for|0.5.14
auto-rename-tag|for|0.1.10
vscode-env|Iro|0.1.0
vscode-gutter-preview|kis|0.30.0
vscode-fix-checksums|leh|1.1.0
inline-parameters|lia|0.2.1
vscode-pastery|max|0.0.4
twig-language-2|mbl|0.9.2
rainbow-csv|mec|3.5.0
php-namespace-resolver|Meh|1.1.9
git-graph|mhu|1.30.0
vscode-checkpoints|mic|1.3.3
vscode-language-pack-hu|MS-|1.48.3
isort|ms-|2022.8.0
python|ms-|2022.20.1
vscode-pylance|ms-|2022.12.20
jupyter|ms-|2022.11.1003412109
jupyter-keymap|ms-|1.0.0
jupyter-renderers|ms-|1.0.12
vscode-jupyter-cell-tags|ms-|0.1.6
vscode-jupyter-slideshow|ms-|0.1.5
remote-ssh|ms-|0.94.0
remote-ssh-edit|ms-|0.84.0
remote-explorer|ms-|0.0.3
vsliveshare|ms-|1.0.5788
color-highlight|nau|2.5.0
vetur|oct|0.36.1
material-icon-theme|PKi|4.23.0
vscode-data-preview|Ran|2.3.0
LiveServer|rit|5.7.9
synthwave-vscode|Rob|0.1.15
vscode-cy-helper|She|1.2.2
highlight-matching-tag|vin|0.10.1
php-debug|xde|1.29.1
markdown-all-in-one|yzh|3.5.0

(3 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
pythonvspyl392:30443607
vserr242:30382549
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vsdfh931cf:30280410
vshan820:30294714
vstes263cf:30335440
pythondataviewer:30285071
vscod805:30301674
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30581797
vsaa593:30376534
pythonvs932:30410667
cppdebug:30492333
vsclangdf:30486550
c4g48928:30535728
dsvsc012cf:30540253
azure-dev_surveyone:30548225
pyindex848cf:30577861
nodejswelcome1cf:30587006
3biah626:30602489
gswce1:30612156
iaj6b796:30613358
89544117:30613380
fim-prod:30623723
vscsbc:30628655

```

</details>

**Comment by lszomoru** [comment]
@bodydisplaynone, the `git.ignoredRepositories` setting that you are highlighting in your screenshot is meant to contain a list of *git repositories* that should be ignored by VS Code when scanning for git repositories. It is not meant to ignore folders in a git repository. To ignore folders in a git repository you would have to use a `.gitignore` file.

**Comment by bodydisplaynone** [comment]
@lszomoru thanks for the reply. My main issue is that I have another machine right here, which has the same project opened and VS sync enabled, so technically, they should behave the same. The .gitignore file in the project does not have the .vscode directory added.

My new machine lists the launch.json as pictured above. My old machine does not, as if it's ignoring the folder properly for some reason. I'm wondering if there is another setting I do not see, maybe a global .gitignore config I added back in the day to exclude this folder from ALL git projects but I don't remember, so every help is appreciated. I also checked my old machine if there is a .gitignore file under userprofile/config/git but I do not.

**Comment by lszomoru** [comment]
@bodydisplaynone, the contents of the "Source Control" view are rendered based on the output of the `git status` command. Could you open a terminal window on both machines, navigate to the folder and execute `git status -uall` in both and see if the output matches. You can also check the value of the `git.untrackedChanges` setting as that also has an impact on how untracked changes are being shown in the "Source Control" view.

**Comment by bodydisplaynone** [comment]
I literally checked all the settings you mentioned. All settings are the same on both machines.

The new machine lists launch.json as untracked when calling git status.
The old machine says no changes.

At this point, I have zero ideas what I did on the old machine to make git "forget" and untrack this file/folder. For now, I added the folder to the .git exclude within the project. The old machine has no entries in this file at all. My idea is still that somehow I added that folder as a global ignore or something, but not sure how or where I did it back in the day.

![image](https://user-images.githubusercontent.com/22660372/208909013-2761324c-944a-49d3-9b90-c918d61eee84.png)

Thanks again for your help and being so responsive! Because it's not really a bug according to the above, more like a git-related question or setting, I'm closing this issue.
